src/lib/alloc.c

```diff
@@ -134,31 +127,35 @@ static void init_heap_map(struct mm_heap *heap, int count)
 		/* init the map[0] */
 		current_map = &heap[i].map[0];
 		current_map->base = heap[i].heap;
-		writeback_block_map(current_map);
+		platform_shared_commit(current_map, sizeof(*current_map));
 
 		/* map[j]'s base is calculated based on map[j-1] */
 		for (j = 1; j < heap[i].blocks; j++) {
 			next_map = &heap[i].map[j];
 			next_map->base = current_map->base +
 				current_map->block_size *
 				current_map->count;
+			platform_shared_commit(next_map, sizeof(*next_map));
+			platform_shared_commit(current_map,
+					       sizeof(*current_map));
+
 			current_map = &heap[i].map[j];
-			writeback_block_map(current_map);
 		}
 
-		dcache_writeback_region(&heap[i], sizeof(heap[i]));
+		platform_shared_commit(&heap[i], sizeof(heap[i]));
```

Comment:
I assume the memory map is overall also protected by a spinlock so that this commit won't break other cores and other commits won't mess up this core. IIRC it was but I want to make sure.

Reply:
@paulstelian97 Yues.